### DISCUSSION TITLE
XML

### DISCUSSION THREAD

**Discussion by 0Grit** [Open-CMSIS-Pack/Open-CMSIS-Pack-Spec#15] [General]
Any discussion yet about switching from XML to another markup language?

(yaml, toml, etc.)?

**Comment by ReinhardKeil** [comment]
Yes, there are discussions about that. Have you seen #12 and the latest demo on PackGen that uses an *.YAML input file to convert CMake->Pack information.

**Reply by iomint** [reply]
@loverdeg: about "switching from XML", is you question investigating a complete switch for open-CMSIS of XML to another markup language including for instance for .pdsc files ? or specific complementary usage to the existing like indeed in the proposal in #12. In the former case, what would be the benefit ? and in the latter do you have use-cases in mind ?

**Reply by 0Grit** [reply]
@ilg-ul

**Reply by 0Grit** [reply]
@iomint that's a good point. I'd think ideally .yaml could be interchangeable for all .pdsc files but I can see how .yaml would be more useful for scenarios like project descriptions versus say a .pdsc for a target MCU.
